src/base-input.js

```diff
@@ -692,27 +708,27 @@ export default class BaseInput extends LitElement {
        */
       if ((!this.value || this.value.length === 0) && this.required) {
         this.validity = 'valueMissing';
-        this.setCustomValidity = this.setCustomValidityValueMissing;
+        this.setCustomValidity = this.setCustomValidityValueMissing || '';
```

Comment:
Not a comment for review, but what does adding the `| | ''` do? I see it all throughout the code when you are setting `this.setCustomValidity`.

Reply:
It means that `this.setCustomValidity` will be an empty string if `this.setCustomValidityMissing` is undefined.

Reply:
The `||` is an `or` operator. E.g. `this.setCustomValidity` will equal the value of `this.setCustomValidityValueMissing` OR an empty string.